index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+const async = require('async');
+
+module.exports = (tape, events) => {
+  let beforeCalled = false;
+  let previousResult = undefined;
+  let outstandingTestCounter = 0;
+  const callTest = (testDescription, testMethod) => {
+    outstandingTestCounter++;
+    async.autoInject({
+      // handle before:
+      before(done) {
+        if (!events.before) {
+          return done();
+        }
+        // if before hasn't been called before:
+        if (!beforeCalled) {
```

Comment:
!events.before || beforeCalled.  you can put both logic in previous if statement